Failing test(s): TestAccDialogflowCXTestCase_dialogflowcxTestCaseFullExample

### Impacted tests

TestAccDialogflowCXTestCase_dialogflowcxTestCaseFullExample


### Affected Resource(s)

google_dialogflow_cx_test_case

### Failure rates

- GA: 100%
- Beta: 100%


### Message(s)

```
=== RUN   TestAccDialogflowCXTestCase_dialogflowcxTestCaseFullExample
=== PAUSE TestAccDialogflowCXTestCase_dialogflowcxTestCaseFullExample
=== CONT  TestAccDialogflowCXTestCase_dialogflowcxTestCaseFullExample
    resource_dialogflow_cx_test_case_generated_test.go:40: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_dialogflow_cx_test_case.basic_test_case will be updated in-place
          ~ resource "google_dialogflow_cx_test_case" "basic_test_case" {
                id               = "projects/ci-test-project-nightly-ga/locations/global/agents/cb41d817-2833-4dbb-8d10-b296b6e473e1/testCases/d9859454-37a6-45f5-bef5-67d298d464fc"
                name             = "d9859454-37a6-45f5-bef5-67d298d464fc"
                tags             = [
                    "#tag1",
                ]
                # (5 unchanged attributes hidden)
        
              ~ test_case_conversation_turns {
                  ~ virtual_agent_output {
                      - session_parameters = jsonencode({})
        
                        # (2 unchanged blocks hidden)
                    }
        
                    # (1 unchanged block hidden)
                }
              ~ test_case_conversation_turns {
                  ~ virtual_agent_output {
                      - session_parameters = jsonencode({})
        
                        # (1 unchanged block hidden)
                    }
        
                    # (1 unchanged block hidden)
                }
        
                # (2 unchanged blocks hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccDialogflowCXTestCase_dialogflowcxTestCaseFullExample (16.57s)
FAIL

```


### Test Debug Log

- [link](https://storage.cloud.google.com/teamcity-logs/nightly/TerraformProviders_GoogleCloud_GOOGLE_NIGHTLYTESTS/2025-02-20/cdc6990/debug-google-cdc6990-326101-TestAccDialogflowCXTestCase_dialogflowcxTestCaseFullExample.txt)

--- a/mmv1/products/dialogflowcx/TestCase.yaml
+++ b/mmv1/products/dialogflowcx/TestCase.yaml
@@ -35,6 +35,7 @@ custom_code:
   pre_read: 'templates/terraform/pre_create/dialogflow_set_location.go.tmpl'
   pre_update: 'templates/terraform/pre_create/dialogflow_set_location.go.tmpl'
   pre_delete: 'templates/terraform/pre_create/dialogflow_set_location.go.tmpl'
+  constants: 'templates/terraform/constants/dialogflowcx_test_case_session_parameters_diff_suppress.go.tmpl'
 exclude_sweeper: true
 examples:
   - name: 'dialogflowcx_test_case_full'
@@ -194,6 +195,7 @@ properties:
               description: |
                 The session parameters available to the bot at this point.
               state_func: 'func(v interface{}) string { s, _ := structure.NormalizeJsonString(v); return s }'
+              diff_suppress_func: 'dialogflowcxTestCaseSessionParametersDiffSuppress'
               custom_flatten: 'templates/terraform/custom_flatten/json_schema.tmpl'
               custom_expand: 'templates/terraform/custom_expand/json_schema.tmpl'
               validation:

--- a/mmv1/products/dialogflowcx/TestCase.yaml
+++ b/mmv1/products/dialogflowcx/TestCase.yaml
@@ -42,9 +42,6 @@ examples:
     primary_resource_id: 'basic_test_case'
     vars:
       agent_name: 'dialogflowcx-agent'
-    # Skip in VCR until the test issue is resolved
-    # https://github.com/hashicorp/terraform-provider-google/issues/21532
-    skip_vcr: true
 parameters:
   - name: 'parent'
     type: String

--- a/mmv1/templates/terraform/constants/dialogflowcx_test_case_session_parameters_diff_suppress.go.tmpl
+++ b/mmv1/templates/terraform/constants/dialogflowcx_test_case_session_parameters_diff_suppress.go.tmpl
@@ -1,0 +1,22 @@
+func dialogflowcxTestCaseSessionParametersDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
+	// Treat empty string and empty JSON object as equivalent
+	if (old == "" || old == "{}") && (new == "" || new == "{}") {
+		return true
+	}
+
+	if old == "" || new == "" {
+		return old == new
+	}
+
+	var oldJson, newJson interface{}
+
+	if err := json.Unmarshal([]byte(old), &oldJson); err != nil {
+		return false
+	}
+
+	if err := json.Unmarshal([]byte(new), &newJson); err != nil {
+		return false
+	}
+
+	return reflect.DeepEqual(oldJson, newJson)
+}

